mne/io/edf/edf.py

```diff
@@ -1693,10 +1979,16 @@ def read_raw_edf(
     The EDF specification allows storage of subseconds in measurement date.
     However, this reader currently sets subseconds to 0 by default.
     """
-    input_fname = os.path.abspath(input_fname)
-    ext = os.path.splitext(input_fname)[1][1:].lower()
-    if ext != "edf":
-        raise NotImplementedError(f"Only EDF files are supported, got {ext}.")
+    if not _file_like(input_fname):
+        input_fname = os.path.abspath(input_fname)
+        ext = os.path.splitext(input_fname)[1][1:].lower()
```

Comment:
Better to prefer pathlib interface, which `_check_fname` will provide for you 
```
input_fname = _check_fname(fname=input_fname, overwrite="read", must_exist=True)
```
now `input_fname` is a `pathlib.Path` and you can get the extension using a method